app/gui2/src/components/GraphEditor/GraphNode.vue

```diff
@@ -191,6 +192,28 @@ function getRelatedSpanOffset(domNode: globalThis.Node, domOffset: number): numb
   }
   return 0
 }
+
+const timeBetweenClicks = 200
+let lastClickTime = 0
+let clickCount = 0
+let singleClickTimer: ReturnType<typeof setTimeout>
+function handlePortClick() {
+  clickCount++
+  if (clickCount === 1) {
+    singleClickTimer = setTimeout(function () {
+      clickCount = 0
+      // If within proper time range, consider it as fast clicks
+      if (Date.now() - lastClickTime >= timeBetweenClicks) {
+        emit('outputPortClick')
+      }
+      lastClickTime = Date.now()
+    }, timeBetweenClicks)
+  } else if (clickCount === 2) {
+    clearTimeout(singleClickTimer)
+    clickCount = 0
+    emit('outputPortDoubleClick')
+  }
+}
```

Comment:
A double-clicking helper could be abstracted out of this--I'm sure we'll have use for it elsewhere.